include/mqtt/endpoint.hpp

```diff
@@ -7948,9 +7949,8 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
     >::type
     async_unsuback_imp(
         packet_id_t packet_id,
-        std::uint8_t qos, Args&&... args) {
-        std::vector<std::uint8_t> params;
-        async_send_unsuback(std::move(params), packet_id, qos, std::forward<Args>(args)..., async_handler_t());
+        Args&&... qos) {
+        async_send_unsuback(std::vector<std::uint8_t>({qos...}), packet_id, async_handler_t());
```

Comment:
Do parameter pack expansion to create the vector directly. No need for recursion.